components/tidb_query/src/rpn_expr/impl_op.rs

```diff
@@ -130,6 +130,15 @@ fn left_shift(lhs: &Option<Int>, rhs: &Option<Int>) -> Result<Option<Int>> {
     })
 }
 
+#[rpn_fn]
+#[inline]
+fn right_shift(lhs: &Option<Int>, rhs: &Option<Int>) -> Result<Option<Int>> {
```

Comment:
TiDB's implementation is u64 u64 wrapping right shift:

```go
func (b *builtinRightShiftSig) evalInt(row chunk.Row) (int64, bool, error) {
	arg0, isNull, err := b.args[0].EvalInt(b.ctx, row)
	if isNull || err != nil {
		return 0, isNull, err
	}
	arg1, isNull, err := b.args[1].EvalInt(b.ctx, row)
	if isNull || err != nil {
		return 0, isNull, err
	}
	return int64(uint64(arg0) >> uint64(arg1)), false, nil
}
```

Please fix both `right_shift` and `left_shift` implementation.

Reply:
`fn checked_shr(self, rhs: u32) -> Option<Self>`

The `rhs` must be `u32`, can not be `u64`

Reply:
So it means you simply cannot use `checked_shr` because the intended behaviour is a wrapping shr instead of a checked shr.